devchat.go

```diff
@@ -45,19 +46,66 @@ var (
 	rtm       *slack.RTM
 	client    = loadTwitterClient()
 
-	red      = color.New(color.FgHiRed)
-	green    = color.New(color.FgHiGreen)
-	cyan     = color.New(color.FgHiCyan)
-	magenta  = color.New(color.FgHiMagenta)
-	yellow   = color.New(color.FgHiYellow)
-	blue     = color.New(color.FgHiBlue)
-	black    = color.New(color.FgBlack)
-	white    = color.New(color.FgHiWhite)
-	colorArr = []*color.Color{yellow, cyan, magenta, green, white, blue, red}
-
+	red         = color.New(color.FgHiRed)
+	green       = color.New(color.FgHiGreen)
+	cyan        = color.New(color.FgHiCyan)
+	magenta     = color.New(color.FgHiMagenta)
+	yellow      = color.New(color.FgHiYellow)
+	blue        = color.New(color.FgHiBlue)
+	black       = color.New(color.FgBlack)
+	white       = color.New(color.FgHiWhite)
 	devbot      = "" // initialized in main
 	startupTime = time.Now()
 
+	colorstyles = []colorstyle{
+		{"white", buildStyleApplicator(*white)},
+		{"red", buildStyleApplicator(*red)},
+		{"green", buildStyleApplicator(*green)},
+		{"cyan", buildStyleApplicator(*cyan)},
+		{"magenta", buildStyleApplicator(*magenta)},
+		{"yellow", buildStyleApplicator(*yellow)},
+		{"blue", buildStyleApplicator(*blue)},
+		{"black", buildStyleApplicator(*black)},
+		{"easter", buildStyleApplicator(*color.New(color.BgMagenta, color.FgHiYellow))},
+		{"baby", buildStyleApplicator(*color.New(color.BgBlue, color.FgHiMagenta))},
+		{"hacker", buildStyleApplicator(*color.New(color.FgHiGreen, color.BgBlack))},
+		//TODO: find a way to support special colors which rely on the user object
```

Comment:
Great point! Thought about this a bit and I realized that we can simply have the applicator print the username directly in the color to be added without first stripping its own colors. Then we could remove the extra ANSI reset code at the end.